pkg/backend/watch.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/rjeczalik/notify"
+
+	"github.com/pulumi/pulumi/pkg/apitype"
+	"github.com/pulumi/pulumi/pkg/backend/display"
+	"github.com/pulumi/pulumi/pkg/diag/colors"
+	"github.com/pulumi/pulumi/pkg/operations"
+	"github.com/pulumi/pulumi/pkg/util/logging"
+	"github.com/pulumi/pulumi/pkg/util/result"
+)
+
+// Watch watches the project's working directory for changes and automatically updates the active
+// stack.
+func Watch(ctx context.Context, b Backend, stack Stack, op UpdateOperation, apply Applier) result.Result {
+
+	opts := ApplierOptions{
+		DryRun:   false,
+		ShowLink: false,
+	}
+
+	startTime := time.Now()
+
+	go func() {
+		shown := map[operations.LogEntry]bool{}
+		for {
+			logs, err := b.GetLogs(ctx, stack, op.StackConfiguration, operations.LogQuery{
```

Comment:
I wonder if we should plumb these through the usual display event pipe as a new event type (or as stdout events).

Reply:
Yes - directionally that makes sense - but I'll hold off on that to do as part of broader `pulumi logs` refactoring.